parser/src/ast/mod.rs

```diff
@@ -50,11 +50,55 @@ pub enum SourceSection {
 // TRACE
 // ================================================================================================
 
-/// [TraceCols] contains the main and auxiliary trace columns of the execution trace.
+/// [Trace] contains the main and auxiliary trace columns of the execution trace.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Trace {
+    pub main_cols: Vec<TraceCols>,
+    pub aux_cols: Vec<TraceCols>,
+}
+
+/// [TraceCols] is used to represent a single or a group of columns in the execution trace. For
+/// single columns, the size is 0. For groups, the size is the number of columns in the group.
 #[derive(Debug, Eq, PartialEq)]
 pub struct TraceCols {
-    pub main_cols: Vec<Identifier>,
-    pub aux_cols: Vec<Identifier>,
+    name: Identifier,
+    size: u64,
+}
+
+impl TraceCols {
+    pub fn new(name: Identifier, size: u64) -> Self {
+        Self { name, size }
+    }
+
+    pub fn name(&self) -> &Identifier {
+        &self.name
+    }
+
+    pub fn size(&self) -> u64 {
+        self.size
+    }
+}
+
+/// [TraceAccess] is used to represent a column inside a trace column group at the specified index
+/// within the group. For columns outside a group, the index is always 0.
```

Comment:
nit: I think this is a bit confusing. Maybe something like the following would be clearer:

"[TraceAccess] is used to indicate a column in the trace by specifying its index within a set of [TraceCols] with the given identifier. If the identifier refers to a single column ([TraceCols] with size 1), then the index is always zero."